### ISSUE TITLE
Pileup alleles demo

### ISSUE THREAD

**Issue by david4096** [BD2KGenomics/ga4gh-integration#11]
Using the reads functionality it should be possible to provide the allele pileup for a specific position in the genome.

Looking at the distribution of alleles at a single genomic position is useful for some types of variant calling.

A pileup function would accept a genomic position, a set of read group IDs, and would return the alleles at that position for each read group ID.

Here is some pseudocode for what this function ought to do. Beware off-by-one errors.

```
function pileup(contig, position, rgIdArray)
  alleles = {}
  for rgId in rgIdArray
    for read in searchReads(contig, position, position+1, rgId)
      allele = read.alignedSequence[position - read.alignment.position.position]
      alleles[read.id] = {allele: allele, readGroupId: rgId} # store some metadata about the found allele
  return alleles
```

**Comment by achave11-ucsc** [comment]
if on negative strand we need to reverted to the positive strand. position is always the start as read from left to right.

**Comment by achave11-ucsc** [comment]
contig => chromosome

**Comment by david4096** [comment]
When an alignment is placed on the negative strand the reverse complement must be taken before comparing it to a reference. http://www.bugaco.com/calculators/dna_reverse_complement.php

**Comment by david4096** [comment]
https://github.com/david4096/bioapi-examples/pull/6

**Comment by kozbo** [comment]
bump. is this complete?

**Comment by kozbo** [comment]
bump again

**Comment by david4096** [comment]
Yes, the above PR fulfilled it! I haven't recently tested it though.
